cmp highlight support

micke opened the pull request:
This PR adds support for new highlight groups introduced in cmp: https://github.com/hrsh7th/nvim-cmp#highlights

Example:

![image](https://user-images.githubusercontent.com/22270/143958342-233f7e86-f5a9-4ab5-b68e-382bae6d6de4.png)

EdenEast commented:
Thank you for the PR! 

I did check to see what the theme file sets for all these components. I did see that functions and methods are set to blue. I do like having the match style as blue so setting these to pink would be my alternative color.

I only made one comment on the change and then we can get this merged.

micke commented:
I tried to match the kind colors to the existing colors set by the theme but thought that the color for function and variable where too close.

But I agree, I've updated the color for function and method, removed bold from Match so that it's more in line with for example Telescope and added some more colors.

EdenEast commented:
LGTM!
